pkg/bpf/bpf.go

```diff
@@ -101,6 +101,13 @@ func EnableMapPreAllocation() {
 	atomic.StoreUint32(&preAllocateMapSetting, 0)
 }
 
+// DisableMapPreAllocation disables BPF map pre-allocation as a default
+// setting. Some map types enforces pre-alloc strategy so this does not
+// take effect in that case.
```

Comment:
Might be worth noting here as well that this only affects the creation of new maps, not existing maps.

Reply:
Ok, will push a fix later.

Reply:
Done (https://github.com/cilium/cilium/pull/7538/commits/8280912414b4b50800a879c2582e80a4fe08eb54)